### ISSUE TITLE
Why does netlify git-gateway use oauth instead of app?

### ISSUE THREAD

**Issue by rens-io** [netlify/git-gateway#35]
I really like that I can connect netlify with my github account through the github app instead of giving full access to all my repos.

When I wish to use netlify git-gateway though it asks to connect with oauth instead of app. I don't like giving full access to all my public and private repos. Is it not possible to use app instead of oauth for github so I can select just the repo I want to give netlify access to?

**Comment by scientist1642** [comment]
Yeah the same question, netlify looks super friendly and all, but this step still made me uneasy.  Not familiar with github api - is it not able to provide required permissions for one repo only?

**Comment by ssfivy** [comment]
I found this netlify forum post that seems to explains the rationale [here](https://community.netlify.com/t/git-gateway-unnecessary-request-for-github-authorisation/743).

Seems rather unfortunate since I was following up [this hugo + academic + netlify guide](https://georgecushen.com/create-your-website-with-hugo/) to make some basic website and being asked to give all repository access was a bit more than I expected!

Would be interested in finer grained permission for this.
